Sources/Subs.php

```diff
@@ -899,6 +899,28 @@ function permute($array)
 	return $orders;
 }
 
+// Rearranges all parameters to be in the right order.  Returns TRUE if no parameters are leftover.
+function_param_order($message, &$parameters, &$out)
+{
+	$params = explode(' ', substr($message, 0, strpos($message, ']')));
```

Comment:
If I were to check for the existance of an opening bracket, and scan for the closing bracket, then look for yet another closing bracket, do you think that would solve this problem?